### DISCUSSION TITLE
Added: LDM notes for August 20, 2018

### DISCUSSION THREAD

**Discussion by agocke** [dotnet/csharplang#1836] [General]
https://github.com/dotnet/csharplang/blob/62185017405127218e80274f62fb5fd596e4f133/meetings/2018/LDM-2018-07-20.md

## Agenda

Nullable open issues:

1. Remaining questions on [suppression operator](https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fdotnet%2Froslyn%2Fissues%2F28271&data=02%7C01%7C%7C6defe1e21ab54cce8d0008d606be5d23%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636703812006445395&sdata=DAdh5dev1mnr%2F5zxtvuJVcHP%2Bzewrzz4z9iuGkl%2BUHg%3D&reserved=0) (and possibly cast)
2. Does a dereference update the null-state?
3. Null contract attributes
4. Expanding the feature
5. Is T? where T : class? allowed or meaningful?
6. Confirm that oblivious should be ephemeral
7. Unconstrained `T` in `List<T>` then `FirstOrDefault()`. What attribute to annotate `FirstOrDefault`?

**Comment by orthoxerox** [comment]
Is it possible for you to share the list of null contract attributes there?

**Comment by agocke** [comment]
@jcouv I believe you had the full list. Could you link that?

**Comment by jcouv** [comment]
The list of attributes is not finalized yet. The issue tracking this has a good number of ideas: https://github.com/dotnet/roslyn/issues/26761

**Comment by nil4** [comment]
@agocke shouldn't the link be [LDM-2018-**08-20**.md](https://github.com/dotnet/csharplang/blob/master/meetings/2018/LDM-2018-08-20.md), rather than `07-20`

**Comment by agocke** [comment]
Yup, typo. I just updated this, but the content is the same, so I'll keep the current link.
